src/components/Gallery/hooks/useImageZoom/useImageZoomDesktop.ts

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+
+import {useLatest} from '../useLatest';
+
+import {DRAG_THRESHOLD, MAX_SCALE_DESKTOP, MIN_SCALE} from './constants';
+import type {Position, ZoomActions, ZoomConstraints, ZoomState} from './types';
+
+interface UseImageZoomDesktopProps {
+    enabled: boolean;
+    zoomState: ZoomState;
+    zoomActions: ZoomActions;
+    constraints: ZoomConstraints;
+}
+
+interface UseImageZoomDesktopReturn {
+    handlers: {
+        onClick?: (e: React.MouseEvent) => void;
+        onMouseDown?: (e: React.MouseEvent) => void;
+    };
+    isDragging: boolean;
+}
+
+/**
+ * Desktop-specific zoom logic using mouse events
+ *
+ * Features:
+ * - Click to toggle between 1x and 2x zoom
+ * - Drag to pan when zoomed
+ * - Click vs drag detection (3px threshold)
+ * - Document-level mouse listeners for smooth drag
+ *
+ * @param props - Configuration with zoom state, actions, and constraints
+ * @returns Handlers and dragging state
+ */
+export function useImageZoomDesktop(props: UseImageZoomDesktopProps): UseImageZoomDesktopReturn {
+    const {enabled, zoomState, zoomActions, constraints} = props;
+    const {scale, position} = zoomState;
+    const {setScale, setPosition, resetZoom} = zoomActions;
+    const {imageSizeRef, containerSizeRef, constrainPosition, imageFitsContainer} = constraints;
+
+    // Use ref to get current value of imageFitsContainer in callbacks
+    const imageFitsContainerRef = useLatest(imageFitsContainer);
+
+    // Desktop-specific state
+    const [isDragging, setIsDragging] = React.useState(false);
+    const [dragStart, setDragStart] = React.useState<Position | null>(null);
+    const [hasMoved, setHasMoved] = React.useState(false);
+
+    /**
+     * Handle image click - toggle zoom
+     * Only toggle if there was no drag movement
+     */
+    const handleImageClick = React.useCallback(
+        (e: React.MouseEvent) => {
+            if (!enabled) {
+                return;
+            }
+
+            e.stopPropagation();
```

Comment:
This can break closing popups logic (if we render the dropdown in the top right corner of the gallery)